### ISSUE TITLE
Document how init.ps1 scripts should find paths to other extracted packages

### ISSUE THREAD

**Issue by joelverhagen** [NuGet/Home#3219]
Today we have no guidance of how `init.ps1` scripts should find other packages.

Suppose a package (`packageA`) has a dependency (`packageB`) as well as an `init.ps1`. Additionally, suppose that `init.ps1` needs to execute an assembly found in that dependency. When `packageA` is pulled into a packages.config project, the following structure is laid down on disk.

```
.\MySolution.sln
.\MyProject\MyProject.csproj
.\MyProject\packages.config
.\packages\packageA.1.0.0\packageA.nuspec
.\packages\packageA.1.0.0\tools\init.ps1
.\packages\packageB.1.0.0\packageB.nuspec
.\packages\packageB.1.0.0\lib\net45\packageB.dll
```

A naive implementation of an `init.ps1` that loads `packageB.dll` would take the input `$installPath`, go up a directory, and append the relative path `packageB.1.0.0\lib\net45\packageB.dll`.

Unfortunately, this does not work if the `init.ps1` is running inside of the user packages folder, which is the case for project.json projects. The folder structure of the user packages folder looks something like this (notice the 2-level folder structure, instead of 1-level):

```
~\.nuget\packages\packageA\1.0.0\packageA.nuspec
~\.nuget\packages\packageA\1.0.0\tools\init.ps1
~\.nuget\packages\packageB\1.0.0\packageB.nuspec
~\.nuget\packages\packageB\1.0.0\lib\net45\packageB.dll
```

In short, we do not want package authors to bake NuGet folder structure knowledge into their scripts. This prevents us from changing this folder structure in the future and leads to complex `init.ps1` scripts that have to haphazardly detect V2 vs. V3 folder structure... or just not work in some scenarios.

We should document the "right" way for a script to do this. Options:
1. Provide a PowerShell API to get the base path to an extracted folder.
2. Document how an `init.ps1` in `packageB` could export information that `packageA` `init.ps1` could leverage.
3. Others?

**Comment by PatoBeltran** [comment]
At this time we suggest moving from packages.config to PackageReference with VS 2017 to resolve your issue. You can find out more about this here: https://blog.nuget.org/20170316/NuGet-now-fully-integrated-into-MSBuild.html

**Comment by joelverhagen** [comment]
@PatoBeltran, isn't `init.ps1` supported in PackageReference?

**Comment by PatoBeltran** [comment]
@joelverhagen you are correct, my bad. I will reopen this issue as it was closed by mistake.
